sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -61,29 +58,72 @@ public URI getRootUri() {
         return this.uriHelper.getRootUri();
     }
 
-    HttpRequests getHttpClient() {
-        return httpClient;
-    }
+    // - if 2xx then return stream
+    // - map 404 to NoResourceException
+    // - if there's a couch error returned as json, unmarshall and throw
+    // - anything else, just throw the IOException back, use the cause part of the exception?
 
-    public List<String> getList(URI uri) {
-        InputStream is = null;
+    // it needs to catch eg FileNotFoundException and rethrow to emulate the previous exception handling behaviour
+    private InputStream executeToInputStream(HttpConnection connection) throws CouchException {
+
+        // all couchclient requests want to receive application/json responses
+        connection.requestProperties.put("Accept", "application/json");
```

Comment:
Are you confusing this with the `content-type` header which is set by the first arg to `Http.POST` etc?

All the `CouchClient` methods which do something with the response body need set this header, and it doesn't do any harm for the ones which don't process the response body. So it's briefer to have this in one helper method than at all the call sites.